### PR TITLE
Shortest Uncommon subseq algo added

### PR DESCRIPTION
**Pull request by somya51p**
### Have you read the [Contributing Guidelines on Pull Requests]
(https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description
 A string of length n has 2^n different possible subsequences, but by using the concept of dynamic programming the time complexity reduces to O(mn^2).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

This is the pull request to issue #3696

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@somya51p Please update the readme file.

**Comment by somya51p** [comment]
@iamrajiv  Changes made in the readme.md, kindly review.

File: C-Plus-Plus/dp/shortest_uncommon_subseq.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    int ans = dp[m][n];
+    if (ans >= MAX)
+        ans = -1;
+
+    return ans;
+}
+
+int main()
+{
+    char S[] = "babab";
```
Comment (iamrajiv) [inline C-Plus-Plus/dp/shortest_uncommon_subseq.cpp]: @somya51p Make this as user input.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/shortest_uncommon_subseq.cpp (hunk 1 above)
Comment (somya51p) [reply C-Plus-Plus/dp/shortest_uncommon_subseq.cpp]: @iamrajiv  Thanks for your review. Made the suggested changes. Have a look.

Comment (iamrajiv) [inline C-Plus-Plus/dp/shortest_uncommon_subseq.cpp]: @somya51p Take input from the user it is still not updated.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/shortest_uncommon_subseq.cpp (hunk 1 above)
Comment (somya51p) [reply C-Plus-Plus/dp/shortest_uncommon_subseq.cpp]: @iamrajiv By mistake, I have reverted the changes. It is fine now. Please review and pardon me for my carelessness.

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@somya51p Please provide the issue number.

**Comment by somya51p** [comment]
This is PR for issue number #3696.

**Comment by iamrajiv** [comment]
@all-contributors please add @somya51p for her code

**Pull request merged**